Devs-AA/Barefoot
# #168662123 Get Trip Stats

#### What does this PR do?
Enables users to get the number of trips made in a specified timeframe

#### Description of Task to be completed?
- Provided a user is logged in, the user should be able to request for the number of trips h/she has made within the specified timeframe
- Ensure only requesters can check trip stats

#### How should this be manually tested?
- git clone this repo using the command `git@github.com:Devs-AA/Barefoot.git`
  - cd to the project
  - checkout to the branch using the command `git checkout ft-trip-stats-168662123`
  - create a `.env` file and set PORT,JWT_SECRET and DB variables as specified in the .env.example file.
  - Run the command `npm install` in the terminal.
  - Open a terminal and run `npm test`.
  - Open a terminal and run `npm run migrate:dev` and `npm run seed:dev`
  1. Open Postman, make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below in the request body.
- `{email: requester2@gmail.com, password: Password1$}`
2. Copy the token in the response
3. Make a GET request to `http:localhost:<port>/api/v1/trips/stats` with the token obtained above in the request header as `authorization`=`Bearer <token obtained>`. The appropriate response will be received. The request body should conform with the image below
![Screenshot (114)](https://user-images.githubusercontent.com/46751363/69835896-20c44b00-1245-11ea-86b1-bee4d5d8b5f8.png)


#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168662123](https://www.pivotaltracker.com/story/show/168662123)

#### Screenshots (if appropriate)

![Screenshot (113)](https://user-images.githubusercontent.com/46751363/69835827-a85d8a00-1244-11ea-8897-f5e35e9f7f6e.png)

#### Questions
N/A
